webui/webui.go

```diff
@@ -0,0 +1,60 @@
+// Package webui provides a method to find the Web UI (happa) URL, based on the installations's API endpoint URL.
+package webui
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	apiStandardHostName = "api"
+
+	webUIStandardHostName = "happa"
+)
+
+// BaseURL returns the web UI base URL derived from an API URL.
+func BaseURL(apiEndpoint string) (string, error) {
+	parsed, err := url.Parse(apiEndpoint)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	hostNameParts := strings.Split(parsed.Hostname(), ".")
+
+	// replace first part with 'happa' if it is 'api'
+	if len(hostNameParts) < 2 {
+		return "", microerror.Mask(unsupportedHostNameError)
+	}
+
+	if hostNameParts[0] != apiStandardHostName {
+		return "", microerror.Maskf(unsupportedHostNameError, "host name must start with 'api'")
```

Comment:
I guess if you take the trouble to make `apiStandardHostName` then you
```suggestion
		return "", microerror.Maskf(unsupportedHostNameError, fmt.Sprintf("host name must start with '%s'", apiStandardHostName))
```